### ISSUE TITLE
Why are auth and apikey config methods needed when it is clearer to set directly on the middleware?

### ISSUE THREAD

**Issue by pajtai** [SolidDigital/bridgetown-api-js#4]
This is especially true for auth tokens, where the middleware field is `authToken`, but the property to set inside the configuration method is `token`.

For example instead of:

``` javascript
bridgetown.configure(function(config){
    config.validate.token(_validateToken);
});

bridgetown.configure(function(config){
    config.validate.apiKey(_validateApiKey);
});
```

Isn't this clearer and more succinct?

``` javascript
bridgetown.middleware.authToken.validate = _validateToken;
bridgetown.middleware.apiKey.validate = _validateApiKey;
```

**Comment by pajtai** [comment]
also, there should be a way to support multiple validate methods. different routers might need different validate methods - for example your ghapi validate methods might be different from your apps validate methods - there should be a function that returns a curried instance of the middleware with the validate method baked in

**Comment by pajtai** [comment]
initial issue and follow up comment closed by 7a5dfa2784bb4d342e2c7ea6d3b4a5dc86639c0d
